src/Snowflake/Command.php

```diff
@@ -0,0 +1,166 @@
+                QueryBuilder::quoteIdentifier($shareName)
+            ));
+
+            $connection->query(sprintf(
+                'GRANT SELECT ON ALL TABLES IN DATABASE %s TO SHARE %s;',
+                QueryBuilder::quoteIdentifier($database),
+                QueryBuilder::quoteIdentifier($shareName)
+            ));
+
+            $connection->query(sprintf(
+                'ALTER SHARE %s ADD ACCOUNT=%s;',
+                QueryBuilder::quoteIdentifier($shareName),
+                $destinationAccount //RL74503.KEBOOLA_BYODB_MIGRATION_APP_3
+            ));
+        }
+    }
+
+    public static function createDatabasesFromShares(
+        Connection $connection,
+        array $databases,
+        string $sourceAccount
+    ): void {
+        foreach ($databases as $database) {
+            $shareDbName = $database . '_SHARE';
+
+            $connection->query(sprintf(
+                'DROP DATABASE IF EXISTS %s;',
+                QueryBuilder::quoteIdentifier($shareDbName)
+            ));
+
+            $connection->query(sprintf(
+                'CREATE DATABASE %s FROM SHARE IDENTIFIER(\'%s.%s\');',
+                QueryBuilder::quoteIdentifier($shareDbName),
+                $sourceAccount,
+                self::MIGRATION_SHARE_PREFIX.$database
+            ));
+        }
+    }
+
+    public static function cloneDatabaseFromShared(Connection $connection, array $databases): void
+    {
+        foreach ($databases as $database) {
+            $shareDbName = $database . '_SHARE';
+
+            $connection->query(sprintf(
+                'DROP DATABASE IF EXISTS %s;',
+                QueryBuilder::quoteIdentifier($database)
+            ));
+
+            $connection->query(sprintf(
+                'CREATE DATABASE %s;',
+                QueryBuilder::quoteIdentifier($database)
+            ));
+
+            $schemas = $connection->fetchAll(sprintf(
+                'SHOW SCHEMAS IN DATABASE %s;',
+                QueryBuilder::quoteIdentifier($shareDbName)
+            ));
+
+            foreach ($schemas as $k => $schema) {
+                if (in_array($schema['name'], self::SKIP_CLONE_SCHEMAS)) {
+                    continue;
+                }
+                $schemaName = $schema['name'];
+
+                $connection->query(sprintf(
+                    'CREATE SCHEMA %s.%s;',
+                    QueryBuilder::quoteIdentifier($database),
+                    QueryBuilder::quoteIdentifier($schemaName)
+                ));
+
+                $tables = $connection->fetchAll(sprintf(
+                    'SHOW TABLES IN SCHEMA %s.%s;',
+                    QueryBuilder::quoteIdentifier($shareDbName),
+                    QueryBuilder::quoteIdentifier($schemaName)
+                ));
+
+                foreach ($tables as $table) {
+                    $tableName = $table['name'];
+
+                    $connection->query(sprintf(
+                        'CREATE TABLE %s.%s.%s AS SELECT * FROM %s.%s.%s;',
```

Comment:
CLONE ze share asi nejde, co? To jsme myslím řešili... https://docs.snowflake.com/en/sql-reference/sql/create-clone.html

Reply:
nejde no... končí to na 
```
Cannot clone from a database that was imported from a share.
```